app/knight_fighting/knight.py

```diff
@@ -0,0 +1,31 @@
+class Knight:
+    def __init__(self, name: str, power: int,
+                 hp: int, armour: list[dict | None],
+                 weapon: dict, potion: dict = None) -> None:
```

Comment:
If you are separating lines in func/method definition - better to have one parameter in one line like this:
```suggestion
    def __init__(
        self, 
        name: str, 
        power: int,
        hp: int, 
        armour: list[dict | None],
        weapon: dict, 
        potion: dict = None
    ) -> None:
```